src/ui/toolbar/MainToolBar.cc

```diff
@@ -258,14 +264,15 @@ void MainToolBar::_setActiveUAS(UASInterface* active)
     // If switching the UAS, disconnect the existing one.
     if (_mav)
     {
-        disconnect(UASMessageHandler::instance(), &UASMessageHandler::textMessageReceived,  this, &MainToolBar::_handleTextMessage);
-        disconnect(_mav, &UASInterface::heartbeatTimeout,                                   this, &MainToolBar::_heartbeatTimeout);
-        disconnect(_mav, &UASInterface::batteryChanged,                                     this, &MainToolBar::_updateBatteryRemaining);
-        disconnect(_mav, &UASInterface::modeChanged,                                        this, &MainToolBar::_updateMode);
-        disconnect(_mav, &UASInterface::nameChanged,                                        this, &MainToolBar::_updateName);
-        disconnect(_mav, &UASInterface::systemTypeSet,                                      this, &MainToolBar::_setSystemType);
-        disconnect(_mav, SIGNAL(statusChanged(UASInterface*,QString,QString)),              this, SLOT(_updateState(UASInterface*,QString,QString)));
-        disconnect(_mav, SIGNAL(armingChanged(bool)),                                       this, SLOT(_updateArmingState(bool)));
+        disconnect(UASMessageHandler::instance(), &UASMessageHandler::textMessageCountChanged,  this, &MainToolBar::_handleTextMessage);
+        disconnect(_mav, &UASInterface::heartbeatTimeout,                                       this, &MainToolBar::_heartbeatTimeout);
+        disconnect(_mav, &UASInterface::batteryChanged,                                         this, &MainToolBar::_updateBatteryRemaining);
+        disconnect(_mav, &UASInterface::modeChanged,                                            this, &MainToolBar::_updateMode);
+        disconnect(_mav, &UASInterface::nameChanged,                                            this, &MainToolBar::_updateName);
+        disconnect(_mav, &UASInterface::systemTypeSet,                                          this, &MainToolBar::_setSystemType);
+        disconnect(_mav, &UASInterface::localizationChanged,                                    this, &MainToolBar::_setSatLoc);
+        disconnect(_mav, SIGNAL(statusChanged(UASInterface*,QString,QString)),                  this, SLOT(_updateState(UASInterface*,QString,QString)));
```

Comment:
This will all go away with my next upgrade of the FactSystem. The FactSystem is meant to support multiple providers. Currently the only provider is parameters. Next provider is mavlink messages. This will replace the myriad of UASInterface signals. So for now I’d leave this alone because it will al go away at some point.

> On Mar 4, 2015, at 1:49 PM, Gus Grubba <notifications@github.com> wrote:
> 
> In src/ui/toolbar/MainToolBar.cc <https://github.com/mavlink/qgroundcontrol/pull/1315#discussion_r25817725>:
> 
> > -        disconnect(UASMessageHandler::instance(), &UASMessageHandler::textMessageReceived,  this, &MainToolBar::_handleTextMessage);
> > -        disconnect(_mav, &UASInterface::heartbeatTimeout,                                   this, &MainToolBar::_heartbeatTimeout);
> > -        disconnect(_mav, &UASInterface::batteryChanged,                                     this, &MainToolBar::_updateBatteryRemaining);
> > -        disconnect(_mav, &UASInterface::modeChanged,                                        this, &MainToolBar::_updateMode);
> > -        disconnect(_mav, &UASInterface::nameChanged,                                        this, &MainToolBar::_updateName);
> > -        disconnect(_mav, &UASInterface::systemTypeSet,                                      this, &MainToolBar::_setSystemType);
> > -        disconnect(_mav, SIGNAL(statusChanged(UASInterface*,QString,QString)),              this, SLOT(_updateState(UASInterface*,QString,QString)));
> > -        disconnect(_mav, SIGNAL(armingChanged(bool)),                                       this, SLOT(_updateArmingState(bool)));
> > +        disconnect(UASMessageHandler::instance(), &UASMessageHandler::textMessageCountChanged,  this, &MainToolBar::_handleTextMessage);
> > +        disconnect(_mav, &UASInterface::heartbeatTimeout,                                       this, &MainToolBar::_heartbeatTimeout);
> > +        disconnect(_mav, &UASInterface::batteryChanged,                                         this, &MainToolBar::_updateBatteryRemaining);
> > +        disconnect(_mav, &UASInterface::modeChanged,                                            this, &MainToolBar::_updateMode);
> > +        disconnect(_mav, &UASInterface::nameChanged,                                            this, &MainToolBar::_updateName);
> > +        disconnect(_mav, &UASInterface::systemTypeSet,                                          this, &MainToolBar::_setSystemType);
> > +        disconnect(_mav, &UASInterface::localizationChanged,                                    this, &MainToolBar::_setSatLoc);
> > +        disconnect(_mav, SIGNAL(statusChanged(UASInterface*,QString,QString)),                  this, SLOT(_updateState(UASInterface*,QString,QString)));
> If you mean statusChanged(), I didn't because it is an overloaded signal (really don't like those).
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mavlink/qgroundcontrol/pull/1315/files#r25817725>.
>